lib/travis/build/templates/header.sh

```diff
@@ -71,20 +73,47 @@ travis_time_finish() {
   return $result
 }
 
-function travis_nanoseconds() {
+travis_nanoseconds() {
   local cmd="date"
   local format="+%s%N"
   local os=$(uname)
 
   if hash gdate > /dev/null 2>&1; then
-    cmd="gdate" # use gdate if available
+    <%# use gdate if available %>
+    cmd="gdate"
   elif [[ "$os" = Darwin ]]; then
-    format="+%s000000000" # fallback to second precision on darwin (does not support %N)
+    <%# fallback to second precision on darwin (does not support %N) %>
+    format="+%s000000000"
   fi
 
   $cmd -u $format
 }
 
+travis_internal_ruby() {
+  if ! type rvm &>/dev/null; then
+    source <%= home %>/.rvm/scripts/rvm &>/dev/null
+  fi
+  local i selected_ruby rubies_array rubies_array_sorted rubies_array_len
+  rubies_array=( $(
+    rvm list strings \
+      | while read -r v; do
+          if [[ ! "${v}" =~ <%= internal_ruby_regex %> ]]; then
+            continue
+          fi
+          v="${v//ruby-/}"
+          v="${v%%-*}"
+          echo "${v//./}_${v}"
```

Comment:
@henrikhodne yes.  I really want to be able to not care about hypothetical future rubies like `2.1.10`.  Really really.